osbs/utils.py

```diff
@@ -340,9 +349,7 @@ def make_name_from_git(repo, branch, limit=53, separator='-', hash_size=5):
 
     repo = ''.join(repo_chars)
     branch = ''.join(branch_chars)
-    name = separator.join([repo, branch, hash_str])
-
-    return name
+    return separator.join(filter(None, (repo, branch, hash_str)))
```

Comment:
filter is needed to handle case where all of repo, or branch, contain invalid characters.

```python
>>> '-'.join(filter(None, [None, '', 'spam', 'bacon']))
'spam-bacon'
>>> '-'.join([None, '', 'spam', 'bacon'])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sequence item 0: expected string, NoneType found
sequence item 0: expected string, NoneType found
>>> '-'.join(['', 'spam', 'bacon'])
'-spam-bacon'
```